test/features/cloudnative/container.go

```diff
@@ -63,11 +63,11 @@ func checkSampleContainer(sampleApp *sample.App, agCrt []byte, trustedCAs []byte
 	}
 }
 
-func AssessOneAgentContainer(builder *features.FeatureBuilder, agCrt []byte, trustedCAs []byte) {
+func AssessOneAgentContainer(builder *features.FeatureBuilder, agCrt func() []byte, trustedCAs []byte) {
 	builder.Assess("certificates are propagated to OneAgent containers", checkOneAgentContainer(agCrt, trustedCAs))
 }
 
-func checkOneAgentContainer(agCrt []byte, trustedCAs []byte) features.Func {
+func checkOneAgentContainer(agCrt func() []byte, trustedCAs []byte) features.Func {
```

Comment:
```suggestion
func checkOneAgentContainer(agCrtFunc func() []byte, trustedCAs []byte) features.Func {
```
wdyt if we name it as agCrtFunc since we changed type, and it's confusing. Why `trustedCAs` is []byte and agCrt is func.

Reply:
Because agCrt is created and read in runtime.
It shouldn't be evaluated when builder.Assess is called but later when the test-step is actually executed.